### PR TITLE
implement fill for connect="finite"

### PR DESCRIPTION
**Pull request by pijyoi**
Implement the fill behavior for `connect="finite"` as defined in #3068

### CODE REVIEW

**Comment by NilsNemitz** [comment]
I have no further comments.   :)

**Comment by pijyoi** [comment]
this PR seems to fix #2312

File: pyqtgraph/graphicsItems/PlotCurveItem.py [hunk 1]
```diff
@@ -869,17 +869,6 @@ def _getFillPathList(self, widget):
                 baseline=None
             )
 
-        if not self.opts['skipFiniteCheck']:
```
Comment (j9ac9k) [inline pyqtgraph/graphicsItems/PlotCurveItem.py:872]: I love that this check on `skipFiniteCheck` is going away!

**Comment by pijyoi** [comment]
An animation to demonstrate this PR as well as #3068
```python
import random
import numpy as np
import pyqtgraph as pg
from pyqtgraph.Qt import QtCore

# pg.setConfigOptions(useOpenGL=True, enableExperimental=True)

pg.mkQApp()
plt = pg.PlotWidget()
plt.show()
data = np.zeros(1000)
data[:-1] = np.nan

curve = pg.PlotCurveItem(data, connect="finite", fillLevel=0.0, brush=(50, 50, 200, 100))
plt.addItem(curve)

def update():
    val = random.gauss(0, 1)
    data[:-1] = data[1:]
    data[-1] = data[-2] + val

    # inject a nan every now and then
    if random.randrange(500) == 0:
        data[-2] = np.nan

    level = np.nanmean(data)
    curve.setData(data, fillLevel=level)

timer = QtCore.QTimer()
timer.timeout.connect(update)
timer.start(0)

pg.exec()
```

**Comment by j9ac9k** [comment]
Love the diff @pijyoi thanks for simplifying one of our most used GraphicsItems.

**Pull request merged**